pkg/source/source.go

```diff
@@ -54,25 +53,53 @@ type SyncingSource interface {
 }
 
 // Kind creates a KindSource with the given cache provider.
-func Kind[T client.Object](cache cache.Cache, object T, handler handler.TypedEventHandler[T], predicates ...predicate.TypedPredicate[T]) SyncingSource {
+func Kind[T client.Object](cache cache.Cache, object T, handler handler.TypedEventHandler[T]) SyncingSource {
 	return &internal.Kind[T]{
-		Type:       object,
-		Cache:      cache,
-		Handler:    handler,
-		Predicates: predicates,
+		Type:    object,
+		Cache:   cache,
+		Handler: handler,
 	}
 }
 
 var _ Source = &channel[string]{}
 
+type channelOptions struct {
+	DestBufferSize int
+}
+
+// ChannelOption is a functional option for configuring a Channel source.
+type ChannelOption func(*channelOptions)
+
+// WithDestBufferSize specifies the buffer size of dest channels.
+func WithDestBufferSize(destBufferSize int) ChannelOption {
+	return func(o *channelOptions) {
+		if destBufferSize <= 0 {
+			return // ignore invalid buffer size
+		}
+
+		o.DestBufferSize = destBufferSize
+	}
+}
+
 // Channel is used to provide a source of events originating outside the cluster
 // (e.g. GitHub Webhook callback).  Channel requires the user to wire the external
 // source (e.g. http handler) to write GenericEvents to the underlying channel.
-func Channel[T any](source <-chan event.TypedGenericEvent[T], handler handler.TypedEventHandler[T], predicates ...predicate.TypedPredicate[T]) Source {
+func Channel[T any](source <-chan event.TypedGenericEvent[T], handler handler.TypedEventHandler[T], options ...ChannelOption) Source {
```

Comment:
Since the predicates are now passed to WithPredicates instead of this function, we can add an options variadic argument and use that to configure eg. the buffer size.